### PR TITLE
Update IsListMethod to include IList

### PR DESCRIPTION
**Pull request by fengzhou-msft**
# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/Mgmt/Decorator/MethodExtensions.cs [hunk 1]
```diff
@@ -33,7 +33,7 @@ public static bool IsListMethod(this RestClientMethod method, [MaybeNullWhen(fal
 
             if (returnType.IsFrameworkType || returnType.Implementation is not SchemaObjectType)
             {
-                if (returnType.FrameworkType == typeof(IReadOnlyList<>))
+                if (returnType.FrameworkType == typeof(IReadOnlyList<>) || returnType.FrameworkType == typeof(IList<>))
```
Comment (ArthurMa1978) [inline src/AutoRest.CSharp/Mgmt/Decorator/MethodExtensions.cs:36]: add a test?

Comment (ArcturusZhang) [inline src/AutoRest.CSharp/Mgmt/Decorator/MethodExtensions.cs:36]: We have a static method `TypeFactory.IsList` to test if a CSharpType is `IList` or `IReadOnlyList`.
Could we switch to that to make here more readable?

File: src/AutoRest.CSharp/Mgmt/Decorator/MethodExtensions.cs [hunk 2]
```diff
@@ -50,7 +50,7 @@ public static bool IsListMethod(this RestClientMethod method, [MaybeNullWhen(fal
         private static ObjectTypeProperty? GetValueProperty(SchemaObjectType schemaObject)
         {
             return schemaObject.Properties.FirstOrDefault(p => p.Declaration.Name == "Value" &&
-                p.Declaration.Type.IsFrameworkType && p.Declaration.Type.FrameworkType == typeof(IReadOnlyList<>));
+                p.Declaration.Type.IsFrameworkType && (p.Declaration.Type.FrameworkType == typeof(IReadOnlyList<>) || p.Declaration.Type.FrameworkType == typeof(IList<>)));
```
Comment (ArcturusZhang) [inline src/AutoRest.CSharp/Mgmt/Decorator/MethodExtensions.cs:53]: Same here
```suggestion
                TypeFactory.IsList(p.Declaration.Type));
```

File: src/AutoRest.CSharp/Mgmt/Decorator/MethodExtensions.cs (hunk 1 above)
Comment (fengzhou-msft) [reply src/AutoRest.CSharp/Mgmt/Decorator/MethodExtensions.cs]: added

Comment (fengzhou-msft) [reply src/AutoRest.CSharp/Mgmt/Decorator/MethodExtensions.cs]: updated

**Pull request merged**